map/src/infoblock/components/graph/RoadAttributesGraphProvider.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import _ from 'lodash';
+import AppContext from '../../../context/AppContext';
+import { addDistanceToPoints, getTrackPoints } from '../../../context/TracksManager';
+import RoadAttributesGraph from './RoadAttributesGraph';
+
+export default function RoadAttributesGraphProvider({ width }) {
+    const ctx = useContext(AppContext);
+
+    const [roadPoints, setRoadPoints] = useState(null);
+
+    useEffect(() => {
+        if (ctx.selectedGpxFile && ctx.currentObjectType === ctx.OBJECT_TYPE_CLOUD_TRACK) {
+            let points = _.cloneDeep(getTrackPoints(ctx.selectedGpxFile));
+            if (points[0].segment) {
+                setRoadPoints(points);
+            } else {
+                setRoadPoints(null);
+            }
+        } else {
+            setRoadPoints(null);
+        }
+    }, [ctx.selectedGpxFile]);
+
+    const data = useMemo(() => {
+        if (roadPoints) {
+            addDistanceToPoints(roadPoints);
+            let segments = [];
+            const routeTypes = roadPoints[0].segment.routeTypes;
+            roadPoints.forEach((p) => {
+                if (p.segment && p.segment.ext.types) {
+                    let seg = p;
+                    seg.ind = _.indexOf(roadPoints, p);
+                    seg['highway'] = 'unknown';
+                    seg['surface'] = 'unknown';
+                    p.segment.ext.types.split(',').forEach((t) => {
+                        const type = routeTypes[t];
+                        if (type.tag === 'highway') {
+                            seg['highway'] = type.value;
+                        }
+                        if (type.tag === 'surface') {
+                            seg['surface'] = type.value;
+                        }
+                    });
+                    segments.push(seg);
+                }
+            });
+            return generateDataSets(segments);
+        }
+    }, [roadPoints]);
+
+    function generatePastelColor() {
+        let R = Math.floor(Math.random() * 127 + 127);
+        let G = Math.floor(Math.random() * 127 + 127);
+        let B = Math.floor(Math.random() * 127 + 127);
+
+        let rgb = (R << 16) + (G << 8) + B;
+        return `#${rgb.toString(16)}`;
+    }
```

Comment:
Colors for track X should be the same after the track was reloaded (or switched off/on or just clicked in menu). Please think about adding custom random() function where you can use init random seed value (JS Math.random not supports random seed).